### PR TITLE
Eth flow/1531 xdai fields missing

### LINKED ISSUE #1531: No modified slippage field in Gnosis chain
Reported and discussed in https://cowservices.slack.com/archives/C0361CDG8GP/p1669719065917469

Reproducible in Development.

1. Enable ETH flow localStorage.setItem('enableEthFlow', '1')
2. Connect to a wallet in Gnosis chain
3. Select an amount of XDAI in the Sell field
4. Select to token
5. Check the fields

**AR**: No modified slippage field (both for regular and expert modes)

**ER**: the  modified slippage field should be displayed.

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Fixes #1531 

# To Test

1. On gchain, select xDai as sell token
* All fields should be displayed like on goerli/mainnet

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @alfetopito , great job!
The only thing that is missing: 
1. No Slippage field in regular mode for regular orders (all networks)
image
2. No changed (not default) slippage at the bottom of the Swap form for regular mode, regular orders (all networks):
image

File: src/custom/lib/hooks/useCurrency/useCurrencyMod.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+/**
+ * Returns a Currency from the currencyId.
+ * Returns null if currency is loading or null was passed.
+ * Returns undefined if currencyId is invalid or token does not exist.
+ */
+export function useCurrencyFromMap(tokens: TokenMap, currencyId?: string | null): Currency | null | undefined {
+  const nativeCurrency = useNativeCurrency()
+  const { chainId, connector } = useWeb3React()
+  const isNative = Boolean(nativeCurrency && ['ETH', 'XDAI'].includes(currencyId?.toUpperCase() || '')) // MOD!!
```
Comment (alfetopito) [inline src/custom/lib/hooks/useCurrency/useCurrencyMod.ts:106]: This is the only change needed to make this work.
Original uniswap code expects the currencyId to be ETH to consider it native

Without that for gchain, there's a chain reaction and we end up without a fee object, which causes some of the fields not be displayed in the UI

**Comment by alfetopito** [comment]
> Hey @alfetopito , great job! The only thing that is missing:
> 
>     1. No Slippage field in regular mode for regular orders (all networks)
>        image
> 
>     2. No changed (not default) slippage at the bottom of the Swap form for regular mode, regular orders (all networks):
>        image

Thanks for catching this Elena!

Fixed

File: src/cow-react/modules/swap/containers/Row/RowSlippage/index.tsx [hunk 2]
```diff
@@ -29,6 +31,9 @@ export function RowSlippage({ allowedSlippage, showSettingOnClick = true }: RowS
     }),
     [allowedSlippage, nativeCurrency, isEthFlow, showSettingOnClick]
   )
+  if ((!isEthFlow && !isExpertMode) || isWrapOrUnwrap) {
+    return null
+  }
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/cow-react/modules/swap/containers/Row/RowSlippage/index.tsx]: Never mind, this broke the behaviour. Reverted

**Review by nenadV91 — APPROVED** [review]

**Review by elena-zh — APPROVED** [review]
@alfetopito , LGTM now, thanks!

**Pull request merged**